pkg/cloudprovider/providers/rackspace/rackspace.go

```diff
@@ -391,3 +467,145 @@ func (os *Rackspace) GetZone() (cloudprovider.Zone, error) {
 
 	return cloudprovider.Zone{Region: os.region}, nil
 }
+
+// Create a volume of given size (in GiB)
+func (rs *Rackspace) CreateVolume(name string, size int, tags *map[string]string) (volumeName string, err error) {
+	return "", errors.New("unimplemented")
+}
+
+func (rs *Rackspace) DeleteVolume(volumeName string) error {
+	return errors.New("unimplemented")
+}
+
+// Attaches given cinder volume to the compute running kubelet
+func (rs *Rackspace) AttachDisk(diskName string) (string, error) {
+	instanceID, err := readInstanceID()
+	if err != nil {
+		return "", err
+	}
+
+	disk, err := rs.getVolume(diskName)
+	if err != nil {
+		return "", err
+	}
+
+	compute, err := rs.getComputeClient()
```

Comment:
The node (compute client) to attach/detach to should be passed in to the attach/detach methods, instead of computed on the fly. The reason being, the attach/detach logic is going to be moved off the kubelet on to a new controller run on master. This controller should be able to specify what node to attach/detach which disk.